vm/src/stdlib/time_module.rs

```diff
@@ -22,36 +22,62 @@ use num_traits::cast::ToPrimitive;
 use chrono::naive::NaiveDateTime;
 use chrono::{Datelike, Timelike};
 
-fn time_sleep(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(vm, args, required = [(seconds, Some(vm.ctx.float_type()))]);
-    let seconds = objfloat::get_value(seconds);
+#[cfg(unix)]
+fn do_sleep(dur: Duration) -> bool {
+    // this is basically std::thread::sleep, but it catches interrupts and we don't want to
+    let secs = dur.as_secs();
+    let nsecs = i64::from(dur.subsec_nanos());
+
+    let mut ts = libc::timespec {
+        tv_sec: std::cmp::min(libc::time_t::max_value() as u64, secs) as libc::time_t,
+        tv_nsec: nsecs,
+    };
+    let res = unsafe { libc::nanosleep(&ts, &mut ts) };
+    res == -1 && nix::errno::errno() == libc::EINTR
+}
+
+#[cfg(not(unix))]
+fn do_sleep(dur: Duration) -> bool {
+    std::thread::sleep(dur);
+    false
+}
+
+fn time_sleep(seconds: PyFloatRef, vm: &VirtualMachine) -> PyResult<()> {
+    let seconds = seconds.to_f64();
     let secs: u64 = seconds.trunc() as u64;
     let nanos: u32 = (seconds.fract() * 1e9) as u32;
     let duration = Duration::new(secs, nanos);
-    thread::sleep(duration);
-    Ok(vm.get_none())
+    let sleep_func = vm.settings.syscall_functions.sleep.unwrap_or(do_sleep);
```

Comment:
See my other comment in PR #1311 about this construction. I think we should not add the `syscall_functions` member.

Reply:
I decoupled this PR from that change, we can look into that separately.